precompiles/erc20/approve.go

```diff
@@ -211,26 +217,49 @@ func (p Precompile) createAuthorization(ctx sdk.Context, grantee, granter common
 	return p.AuthzKeeper.SaveGrant(ctx, grantee.Bytes(), granter.Bytes(), authorization, &expiration)
 }
 
-func (p Precompile) updateAuthorization(ctx sdk.Context, grantee, granter common.Address, amount *big.Int, authorization *banktypes.SendAuthorization) error {
-	authorization.SpendLimit = sdk.Coins{{Denom: p.tokenPair.Denom, Amount: sdk.NewIntFromBigInt(amount)}}
+func (p Precompile) updateAuthorization(ctx sdk.Context, grantee, granter common.Address, amount *big.Int, authorization *banktypes.SendAuthorization, expiration *time.Time) error {
+	found, denomAmount := authorization.SpendLimit.Find(p.tokenPair.Denom)
+	if found {
+		// NOTE: we are first removing the existing amount and then adding the new one
+		authorization.SpendLimit = authorization.SpendLimit.Sub(denomAmount)
```

Comment:
are we checking that this is always positive?

Reply:
✅ The `Sub` method checks this: https://github.com/evmos/cosmos-sdk/blob/7a885823bc4a8a72c41dcd1381cba9819f487349/types/coin.go#L366-L382

Reply:
You can use `SafeSub` to avoid panic